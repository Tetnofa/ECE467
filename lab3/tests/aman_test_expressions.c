// Test various expressions and operators
void testExpressions() {
    int a;
    int b;
    bool c;
    bool d;
    
    a = 10;
    b = 5;
    
    // Arithmetic
    writeInt(a + b);
    writeInt(a - b);
    writeInt(a * b);
    writeInt(a / b);
    
    // Comparison
    c = a > b;
    d = a <= b;
    writeBool(c);
    writeBool(d);
    
    // Boolean logic
    writeBool(c && d);
    writeBool(c || d);
    writeBool(!c);
} 