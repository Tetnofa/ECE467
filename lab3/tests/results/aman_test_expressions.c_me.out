sema: 12:4 : use of undefined identifier writeInt
sema: 13:4 : use of undefined identifier writeInt
sema: 14:4 : use of undefined identifier writeInt
sema: 15:4 : use of undefined identifier writeInt
sema: 20:4 : use of undefined identifier writeBool
sema: 21:4 : use of undefined identifier writeBool
sema: 24:4 : use of undefined identifier writeBool
sema: 25:4 : use of undefined identifier writeBool
sema: 26:4 : use of undefined identifier writeBool
